models.py

```diff
@@ -48,4 +49,30 @@ def forward(self, x, pad_mask):
         x = self.output_norm(x)
         x = self.output(x)
         return torch.squeeze(x)
-    
\ No newline at end of file
+
+class WatsonCrickAttentionLayer(torch.nn.Module):
+
+    def __init__(self, size, score_matrix):
+        super(WatsonCrickAttentionLayer, self).__init__()
+        self.size = size
+        self.score_matrix = torch.nn.Parameter(score_matrix, requires_grad=False)
+
+        # Linear Components
+        self.q_linear = torch.nn.Linear(size, size)
+        self.k_linear = torch.nn.Linear(size, size)
+        self.v_linear = torch.nn.Linear(size, size)
+
+    def forward(self, x):
```

Comment:
which of q, k, v is supposed to be the sequence / bpp matrix?

Reply:
it could be query or key I believe